### ISSUE TITLE
Gdn_DatabaseStructure::query method should not be exposed as public.

### ISSUE THREAD

**Issue by DaazKu** [vanilla/vanilla#3872]
https://github.com/vanilla/vanilla/pull/3605 adds a nice feature that allows to check a threshold before applying an alter table statement.

The problem is that you have to call `query($sql, true)` for the feature to be enabled.

This is impossible when doing to following:
```
Gdn::structure()->table('TABLE NAME')
    ...// Add new columns or whatever
    ->setAlterTableThreshold(100)
    ->set(false, false)
```

Callstack:
```
query($sql, $checkThreshold = false) // this is called like so: query($sql)
_modify($Explicit = false)
set($Explicit = false, $Drop = false)
```

To fix the problem we have 2 choices 
- we propagate the `$checkThreshold` variable through the stack (and everywhere else where is it needed)
- we assume that if `(bool)$this->alterTableThreshold` is true we apply the check.

**Comment by tburry** [comment]
There are some alter commands that check the threshold and some that do not. The idea is that some alters will not put too much load on the database on recent versions of MySQL so they are safe to execute.

**Comment by DaazKu** [comment]
It would be nice to be able to restrict an alterTable on row counts alone.
Let's say I'm creating an index on a table that has a million+ rows, a plugin could use the feature.

**Comment by tburry** [comment]
The reason why I don't do the threshold on indexes is because of [fast index creation](https://dev.mysql.com/doc/refman/5.5/en/innodb-create-index-overview.html). We've not seen indexes be a major bottleneck. This is something that probably can be benchmarked.

**Comment by DaazKu** [comment]
At first I got confused because `query` is public and you can use the second parameter to force an alterTable check but after a more thorough analysis of the code I realized that query should not, in theory, be used outside of its class. (There are only 8 external usages of it through vanilla)

I think that the documentation should be updated to say that the feature check only some alter table changes.
We may want to change `query` from public to protected if we feel nitpicky.

**Comment by tburry** [comment]
There are a lot of legacy methods in these core classes that certainly should be protected. I notice this when I revisit older code. Great point.

**Comment by DaazKu** [comment]
Closed by #4038
